utils/vllm_warmup.py

```
@@ -0,0 +1,257 @@
+# ---------------------------------------------------------------------------
+
+def wait_for_server(llm_url: str, timeout: int = 120) -> bool:
+    """Block until ``GET <llm_url>/models`` returns 200 or *timeout* expires."""
+    check_url = f"{llm_url}/models"
+    for attempt in range(timeout):
+        try:
+            r = requests.get(check_url, timeout=5)
+            if r.status_code == 200:
+                logger.info("Connected to vLLM server at %s", llm_url)
+                return True
+        except Exception as exc:
```

Ruff (0.15.2)

[warning] Do not catch blind exception: `Exception`

(BLE001)